resources/net.go

```diff
@@ -0,0 +1,804 @@
+		return false
+	}
+	if obj.Name != res.Name {
+		return false
+	}
+	if obj.State != res.State {
+		return false
+	}
+	if obj.DHCP != res.DHCP {
+		return false
+	}
+	if (obj.Addrs == nil) != (res.Addrs == nil) {
+		return false
+	}
+	if obj.Addrs != nil && res.Addrs != nil {
+		if len(obj.Addrs) != len(res.Addrs) {
+			return false
+		}
+		objAddrs := obj.Addrs
+		resAddrs := res.Addrs
+		sort.Strings(objAddrs)
+		sort.Strings(resAddrs)
+		for i := range obj.Addrs {
+			if obj.Addrs[i] != res.Addrs[i] {
+				return false
+			}
+		}
+	}
+	if obj.Gateway != res.Gateway {
+		return false
+	}
+
+	return true
+}
+
+// Close cleans up some resources so we can exit cleanly.
+func (obj *NetRes) Close() error {
+	// close the dbus connection
+	obj.sdbus.Close()
+	// close the message bus
+	if err := obj.gdbus.Close(); err != nil {
+		return errwrap.Wrapf(err, "error closing godbus message bus")
+	}
+
+	return obj.BaseRes.Close()
+}
+
+// UnmarshalYAML is the custom unmarshal handler for this struct.
+// It is primarily useful for setting the defaults.
+func (obj *NetRes) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawRes NetRes // indirection to avoid infinite recursion
+
+	def := obj.Default()     // get the default
+	res, ok := def.(*NetRes) // put in the right format
+	if !ok {
+		return fmt.Errorf("could not convert to NetRes")
+	}
+	raw := rawRes(*res) // convert; the defaults go here
+
+	if err := unmarshal(&raw); err != nil {
+		return err
+	}
+
+	*obj = NetRes(raw) // restore from indirection with type conversion!
+	return nil
+}
+
+// unitFileContents builds the unit file contents from the input
+func (obj *NetRes) unitFileContents() []byte {
+	// build the unit file contents
+	u := []string{"[Match]"}
+	u = append(u, fmt.Sprintf("Name=%s", obj.GetName()))
+	u = append(u, "[Network]")
+	if obj.DHCP {
+		u = append(u, "DHCP=yes")
+	}
+	for _, addr := range obj.Addrs {
+		u = append(u, fmt.Sprintf("Address=%s", addr))
+	}
+	if obj.Gateway != "" {
+		u = append(u, fmt.Sprintf("Gateway=%s", obj.Gateway))
+	}
+	c := strings.Join(u, "\n")
+	return []byte(c)
+}
+
+// getAddrsFromIface returns the provided interface's IPv4 and IPv6 addresses.
+// If an address isn't found, an empty string is returned in its place.
+func getAddrsFromIface(name string) ([]net.Addr, error) {
+	// find the interface
+	iface, err := net.InterfaceByName(name)
+	if err != nil {
+		return nil, errwrap.Wrapf(err, "error finding interface: %s", name)
+	}
+
+	// check the interface's ip addresses
+	return iface.Addrs()
+}
+
+// restartSystemdUnit restarts the specified systemd unit.
+func restartSystemdUnit(conn *systemd.Conn, unit, mode string) error {
+	// connect to systemd
+	connChan := make(chan string) // catch result information
+	defer close(connChan)
+
+	// restart unit
+	if _, err := conn.RestartUnit(unit, mode, connChan); err != nil {
+		return errwrap.Wrapf(err, "failed to restart unit")
+	}
+
+	// XXX: will this ever hang?
```

Comment:
you tell me :)

unfortunately, sometimes these sorts of things are difficult to know and occasionally involve reading the upstream code a little. it's good practice, but don't dive too long. if you can't find it in 20 min, move on or ask someone.